static/js/containers/CreatePostPage.js

```diff
@@ -87,20 +87,25 @@ class CreatePostPage extends React.Component<CreatePostPageProps> {
     // we may need to change out the postType under certain conditions
     // if the user switches channels and the new channel isn't LINK_TYPE_ANY,
     // and the current postType is null or a postType which is incompatible with the
-    // new channel, we want to set the postType to be the link_type for the channel
+    // new channel, we want to set the postType to be the link_type for the channel.
+    // If the new channel is LINK_TYPE_ANY and both text and url values are empty,
+    // we want to set the postType to null to display the post type options.
     if (
       postForm &&
       channel &&
       ((prevProps.channel && prevProps.channel.name !== channel.name) ||
         !prevProps.channel) &&
-      channel.link_type !== LINK_TYPE_ANY &&
+      (channel.link_type !== LINK_TYPE_ANY ||
+        (emptyOrNil(postForm.value.text + postForm.value.url) &&
+          !emptyOrNil(postForm.value.postType))) &&
```

Comment:
Shouldn't the added logic just be a check to see if the `postType` value changed? I'm pretty sure you can just compare `prevProps.postForm.value.postType` and `this.props.postForm.value.postType` for that. If that value changes, we don't need to know anything else to determine that the form needs an update, right? 

Also, I would create a named helper function for this new logic. The existing conditional is really verbose and it would help to have a named function to read here instead of yet another comparison.

Reply:
If you start creating a post in a link-only channel, the link field will display by default.  If you then switch to a channel that allows both links and text posts, whether or not the form should update depends on whether anything was entered in that link field yet.  If not, the link field should be hidden and the 'Write something' & 'Share a link' buttons should be shown instead.  Otherwise, the URL entered by the user should continue to display.

Reply:
I put the additional logic into a separate function